### PR TITLE
feat(test): e2e simulation test

### PR DESCRIPTION
**Pull request by wangrunji0408**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

The madsim simulation testing framework is now available on e2e test!

You can have a try with:
```sh
RUST_LOG=info ./risedev stest -p risingwave_simulation
```

All risingwave components + pg client will be running on a simulator.

However, the execution is still **not deterministic**. Eliminating randomness will be our next goal!

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/tests/simulation/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+[target.'cfg(madsim)'.dependencies]
+async-trait = "0.1"
+clap = "3"
+glob = "0.3"
+madsim = "=0.2.0-alpha.5"
+risingwave_compactor = { path = "../../storage/compactor" }
+risingwave_compute = { path = "../../compute" }
+risingwave_frontend = { path = "../../frontend" }
+risingwave_meta = { path = "../../meta" }
+sqllogictest = "0.5.5"
```
Comment (skyzh) [inline src/tests/simulation/Cargo.toml:16]: add workspace_hack here?

**Review by skyzh — APPROVED** [review]
LGTM, good work!

**Review by KveinAxel — APPROVED** [review]
LGTM

**Comment by skyzh** [comment]
Deterministic test will fail and retry for multiple times before merging. Shall we disable retry for now?

**Comment by wangrunji0408** [comment]
> Deterministic test will fail and retry for multiple times before merging. Shall we disable retry for now?

Sure. And the time limit could be relaxed (maybe 8min?)

**Comment by skyzh** [comment]
It's not because of the time limit. It's just because that the test failed, and it started retrying. We should only retry when agent is lost.

**Comment by wangrunji0408** [comment]
Yes. I mean another thing: after this PR introducing e2e simulation test, its execution time exceeded the original limit. We should relax it so that this task won't be killed by timeout.

**Pull request merged**